src/dtclient/processmoduleconfig_test.go

```diff
@@ -284,3 +285,52 @@ func TestAdd(t *testing.T) {
 		})
 	})
 }
+
+func TestProcessModuleConfig_AddProxy(t *testing.T) {
+	const proxy = "dummy-proxy"
+	type fields struct {
+		Revision   uint
+		Properties []ProcessModuleProperty
+	}
+	type args struct {
+		proxy string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   *ProcessModuleConfig
+	}{
+		{
+			name: "add proxy to process module config",
+			fields: fields{
+				Revision:   0,
+				Properties: []ProcessModuleProperty{},
+			},
+			args: args{
+				proxy: proxy,
+			},
+			want: &ProcessModuleConfig{
+				Revision: 0,
+				Properties: []ProcessModuleProperty{
+					{
+						Section: generalSectionName,
+						Key:     "proxy",
+						Value:   proxy,
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			pmc := &ProcessModuleConfig{
+				Revision:   tt.fields.Revision,
+				Properties: tt.fields.Properties,
+			}
+			if got := pmc.AddProxy(tt.args.proxy); !reflect.DeepEqual(got, tt.want) {
```

Comment:
you can probably try to [testify.Equal](https://pkg.go.dev/github.com/stretchr/testify/assert#Equal) instead of reflect. 

https://github.com/Dynatrace/dynatrace-operator/blob/2328a307c407168f5b319c8bd278a92409bab305/.golangci.yml#L133-L135